File: user_spring/src/main/java/com/kjj/user/dto/user/JoinDto.java
package com.kjj.user.dto.user;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class JoinDto {
    private String username;
    private String password;

    public void setEncodePassword(BCryptPasswordEncoder bCryptPasswordEncoder) {
        password = bCryptPasswordEncoder.encode(password);
    }
}

File: user/src/main/kotlin/com/kjj/user/dto/user/JoinDto.kt
package com.kjj.user.dto.user

import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder

data class JoinDto(
    val username: String,
    var password: String
) {
    fun setEncodePassword(encoder: BCryptPasswordEncoder) {
        password = encoder.encode(password)
    }
}